src/App.tsx

```diff
@@ -1,27 +1,59 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
-
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
+
+  const loadGoods = (loadFunction: () => Promise<Good[]>) => {
+    loadFunction().then(goods => {
+      setVisibleGoods(goods);
+    });
+  };
+
+  useEffect(() => {
+    loadGoods(getAll);
+  }, []);
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => {
+          loadGoods(getAll);
+        }}
```

Comment:
```suggestion
        onClick={() => loadGoods(getAll)}
```
same for other buttons.